pkg/operator/ceph/object/user/controller.go

```diff
@@ -310,14 +310,20 @@ func (r *ReconcileObjectStoreUser) initializeObjectStoreContext(u *cephv1.CephOb
 	}
 
 	r.objContext = objContext
-	r.objContext.Endpoint = store.Status.Info["endpoint"]
-	if store.Status != nil && store.Status.Info["secureEndpoint"] != "" {
-		r.objContext.Endpoint = store.Status.Info["secureEndpoint"]
+	if store.Status == nil {
```

Comment:
If nil why not initialize it? How come do we end up in this situation? We should fix it instead of returning an error.

Reply:
To update the object user status, we need to know the endpoint of the object store. So until the object store sets its own status, we cannot continue here. Returning an error would lead to a new reconcile and retry until it is available, so this seems best.

Reply:
I misread that code, I thought this was for `CephObjectStore` but I'm fine with bailing out until `store.Status` is not nil anymore.